### PR TITLE
Show errors on the UI - Sessions Feature

### LINKED ISSUE #36: Handle and Show Api & Network Error on UI

### PR DESCRIPTION
**Pull request by BKinya**
# Scope

_Please make sure to read https://github.com/droidconKE/droidconKE2022Android/docs/CONTRIBUTING.md
and check that you understand and have followed it as best as possible Explain what your feature
does in a short paragraph.
- Added `SessionsUiState` sealed class to help manage different states in the sessions screen: error state, success state, loading state
- Removed ViewModels passed down to individual composables 
- Co-located navigation 

_ please check the below boxes
- [x] I have followed the coding conventions
- [ ] I have added/updated necessary tests
- [x] I have tested the changes added on a physical device
- [x] I have run `./codeAnalysis.sh` to make sure all lint/formatting checks have been done.

## Closes/Fixes Issues
partly fixes #36 

### Screenshots for various Sessions UI state
1. No sessions 2. List of sessions 3. Error screen

Screenshot 2023-03-28 at 01 58 14           Screenshot 2023-03-28 at 01 44 00              Screenshot 2023-03-28 at 02 00 21

### CODE REVIEW

**Comment by BKinya** [comment]
> @BKinya Can you resolve the conflicts? Also since this is a UI feature, can we have a video/screenshot to quickly see

Sure, Let me update

**Comment by kibettheophilus** [comment]
@BKinya check on conflicts too

**Comment by BKinya** [comment]
> @BKinya Can you resolve the conflicts? Also since this is a UI feature, can we have a video/screenshot to quickly see

@wangerekaharun @kibettheophilus I have fixed the conflicts and added screenshots.

**Comment by wangerekaharun** [comment]
@BKinya CI is failing

File: presentation/src/main/java/com/android254/presentation/sessions/components/SessionsErrorComponent.kt [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+            ),
+        )
+        Spacer(
+            modifier = Modifier.height(32.dp)
+        )
+
+        Button(onClick = { retry() }) {
+            Text(
+                modifier = Modifier,
+                text = "Retry",
```
Comment (wangerekaharun) [inline presentation/src/main/java/com/android254/presentation/sessions/components/SessionsErrorComponent.kt:58]: Should be.a string resource

File: presentation/src/main/java/com/android254/presentation/sessions/components/SessionStateComponent.kt [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+                        Modifier.padding(
+                            start = 40.dp,
+                            end = 0.dp,
+                            top = 10.dp,
+                            bottom = 10.dp
+                        )
+                    ) {
+                        Image(
+                            painter = painterResource(id = if (index % 2 == 0) R.drawable.ic_green_session_card_spacer else R.drawable.ic_orange_session_card_spacer),
+                            contentDescription = "spacer icon"
```
Comment (wangerekaharun) [inline presentation/src/main/java/com/android254/presentation/sessions/components/SessionStateComponent.kt:118]: Should be a string resource

File: presentation/src/main/java/com/android254/presentation/sessions/components/SessionStateComponent.kt [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+import com.android254.presentation.models.SessionPresentationModel
+import com.android254.presentation.sessions.models.SessionsUiState
+import com.droidconke.chai.atoms.ChaiDarkGrey
+import com.droidconke.chai.atoms.MontserratRegular
+import com.google.accompanist.swiperefresh.SwipeRefresh
+import com.google.accompanist.swiperefresh.SwipeRefreshState
+import com.google.accompanist.swiperefresh.rememberSwipeRefreshState
+import java.util.*
+
+@RequiresApi(Build.VERSION_CODES.O)
```
Comment (wangerekaharun) [inline presentation/src/main/java/com/android254/presentation/sessions/components/SessionStateComponent.kt:44]: We had a PR that removed this, you should also remove them

**Review by wangerekaharun — COMMENTED** [review]
Looks good, let some few minor comments

File: presentation/src/main/java/com/android254/presentation/sessions/components/SessionStateComponent.kt (hunk 3 above)
Comment (BKinya) [reply presentation/src/main/java/com/android254/presentation/sessions/components/SessionStateComponent.kt]: I've resolved this.

**Review by wangerekaharun — APPROVED** [review]

**Review by jumaallan — APPROVED** [review]

**Pull request merged**